uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Page/AbsenceType.php

```diff
@@ -3,7 +3,7 @@
 use CRM_HRLeaveAndAbsences_Service_AbsenceType as AbsenceTypeService;
 use CRM_HRLeaveAndAbsences_BAO_AbsenceType as AbsenceType;
 
-class CRM_HRLeaveAndAbsences_Page_AbsenceType extends CRM_Core_Page_Basic {
+class CRM_HRLeaveAndAbsences_Page_AbsenceType extends CRM_Core_Page {
```

Comment:
The CRM_Core_Page_Basic is a child of the CRM_Core_Page, I would have thought if we need less functionality, we should have stuck with CRM_Core_Page_Basic class?

Reply:
CRM_Core_Page_Basic extends the functionality of CRM_Core_Page, adding facility for rendering form (using QuickForm). We don't need all the extras since the form is now processed with AngularJS.